packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware.ts

```diff
@@ -195,14 +195,22 @@ const simpleBuyPaymentTypeDictionary = (rawPaymentType: SBPaymentTypes) => {
 
 const getOriginalTimestamp = () => new Date().toISOString()
 
+const id = v4()
+
 const analytics = queuevent<AnalyticsKey, AnalyticsPayload>({
   queueCallback: async (rawEvents) => {
     const res = await fetch('/wallet-options-v4.json')
     const options = await res.json()
 
     const analyticsURL = `${options.domains.api}/events/publish`
 
-    const id = v4()
+    const nabuId = rawEvents.find((event) => event.payload.id)?.payload.id || null
+
+    const context: { traits: { nabu_id: string | null } } = {
```

Comment:
maybe to make type from this one like

```
type ContextType = { traits: { nabu_id: string | null } } 
```

and then use it, but it's really minor